### PR TITLE
Reactivate last survey on sign in

### LINKED ISSUE #1612: Regression: Last active survey not restored after sign-in

### PR DESCRIPTION
**Pull request by gino-m**
Fixes #1612.

@scolsen and/or @shobhitagarwal1612 PTAL?

### CODE REVIEW

File: ground/src/main/java/com/google/android/ground/MainViewModel.kt [hunk 1]
```diff
@@ -74,7 +80,8 @@ constructor(
     return signInState.result.fold(
       {
         when (signInState.state) {
-          SignInState.State.SIGNED_IN -> onUserSignedIn(it!!)
+          SignInState.State.SIGNED_IN ->
+            rxObservable(defaultDispatcher) { send(onUserSignedIn(it!!)) }
```
Comment (shobhitagarwal1612) [inline ground/src/main/java/com/google/android/ground/MainViewModel.kt:84]: What is the purpose of the added constructs? How do we know when to use them?

Comment (shobhitagarwal1612) [reply ground/src/main/java/com/google/android/ground/MainViewModel.kt:84]: The default ktdoc is sufficient. I was mostly concerned about how do we ensure that we apply these changes as we migrate more code to coroutines.

Comment (gino-m) [reply ground/src/main/java/com/google/android/ground/MainViewModel.kt:84]: Makes sense.. Tests won't work correctly without setting the dispatcher, and nothing will be emitted without `send`, so correctness elsewhere can be enforced by unit and manual testing. Lmk if you have another idea!

**Review by shobhitagarwal1612 — APPROVED** [review]
💯

**Pull request merged**